app/renderer/components/publisherToggle.js

```diff
@@ -23,97 +24,89 @@ class PublisherToggle extends ImmutableComponent {
-        verifiedPublisher = !!publisher
-        return false
-      }
-      return true
-    })
-    return verifiedPublisher
+    return this.props.locationInfo.getIn([this.locationId, 'verified'])
   }
 
   get visiblePublisher () {
-    // ledgerPaymentsShown is undefined by default until user decide to permanently hide the publisher
+    // ledgerPaymentsShown is undefined by default until
+    // user decide to permanently hide the publisher,
     // so for icon to be shown it can be everything but false
     const ledgerPaymentsShown = this.hostSettings && this.hostSettings.get('ledgerPaymentsShown')
-    return ledgerPaymentsShown !== false
+    return typeof ledgerPaymentsShown === 'boolean'
+      ? ledgerPaymentsShown
+      : true
   }
 
   get shouldShowAddPublisherButton () {
-    if ((!!this.hostSettings || !!this.validPublisherSynopsis) && this.visiblePublisher) {
-      // Only show publisher icon if ledger is enabled
-      return getSetting(settings.PAYMENTS_ENABLED)
-    }
-    return false
+    return getSetting(settings.PAYMENTS_ENABLED) &&
+      isHttpOrHttps(this.props.location) &&
+      this.visiblePublisher
   }
 
   get l10nString () {
-    if (this.verifiedPublisher && !this.authorizedPublisher) {
-      return 'verifiedPublisher'
-    } else if (this.authorizedPublisher) {
-      return 'enabledPublisher'
+    let l10nData = 'disabledPublisher'
+    if (this.verifiedPublisher && !this.enabledForPaymentsPublisher) {
+      l10nData = 'verifiedPublisher'
+    } else if (this.enabledForPaymentsPublisher) {
+      l10nData = 'enabledPublisher'
     }
-    return 'disabledPublisher'
+    return l10nData
   }
 
   onAuthorizePublisher () {
-    this.authorizedPublisher
-      ? appActions.changeSiteSetting(this.hostPattern, 'ledgerPayments', false)
-      : appActions.changeSiteSetting(this.hostPattern, 'ledgerPayments', true)
+    const hostPattern = getHostPattern(this.publisherId)
+    appActions.changeSiteSetting(hostPattern, 'ledgerPayments', !this.enabledForPaymentsPublisher)
   }
 
   render () {
     return this.shouldShowAddPublisherButton
       ? <span
         data-test-id='publisherButton'
-        data-test-authorized={this.authorizedPublisher}
+        data-test-authorized={this.enabledForPaymentsPublisher}
         data-test-verified={this.verifiedPublisher}
         className={css(styles.addPublisherButtonContainer)}>
         <button
           className={
           css(
             commonStyles.browserButton,
-            !this.authorizedPublisher && this.verifiedPublisher && styles.noFundVerified,
-            this.authorizedPublisher && this.verifiedPublisher && styles.fundVerified,
-            !this.authorizedPublisher && !this.verifiedPublisher && styles.noFundUnverified,
-            this.authorizedPublisher && !this.verifiedPublisher && styles.fundUnverified
+            !this.enabledForPaymentsPublisher && this.verifiedPublisher && styles.noFundVerified,
```

Comment:
if we wanted to make this cleaner, you could do the && for the two conditions above and store in a const, like
```
const showNoFundVerified = !this.enabledForPaymentsPublisher && this.verifiedPublisher
// ...
css(
    showNoFundVerified && styles.noFundVerified,
)
```